catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -742,4 +742,44 @@ private boolean checkSingleFile(Fileset fileset) {
           fileset.name());
     }
   }
+
+  FileSystem getFileSystem(Path path, Map<String, String> config) throws IOException {
+    String defaultFilesystemProvider =
+        (String)
+            propertiesMetadata
+                .catalogPropertiesMetadata()
+                .getOrDefault(config, HadoopCatalogPropertiesMetadata.DEFAULT_FS);
+
+    Map<String, String> newConfig = Maps.newHashMap(config);
+    if (path == null) {
+      if (defaultFilesystemProvider != null) {
+        return getFileSystemByScheme(defaultFilesystemProvider, newConfig);
+      } else {
+        LOG.warn("The path and default filesystem provider are both null, using local file system");
+        return getFileSystemByScheme(LOCAL_FILE_SCHEMA, newConfig);
+      }
+    }
+
+    // Path is not null;
+    if (path.toUri().getScheme() == null) {
+      LOG.warn(
+          "Can't get schema from path: {} and default filesystem provider are both null, using"
+              + " local file system",
+          path);
+      return getFileSystemByScheme(LOCAL_FILE_SCHEMA, newConfig);
+    } else {
+      newConfig.put(DEFAULT_FS, path.toUri().toString());
```

Comment:
What's the purpose of doing this?

Reply:
Pass the `defaultFs` information to the fileSystem provider, for GCS or S3, we need to specify the bucket name, for HDFS we need to provider HDFS namenode address.

Reply:
for exmaple, if the `path` is `hdfs://xxxx/xxx` and `defaultFSProvider` is not set, we can refer the schema from path is `HDFS`, however, if we want to get a FileSystem instance, then we should explicitly set `fs.defaultFS` to a HDFS path in `FileSystem.getinstance()`